tests/Unit/DataStructures/Test_OrientVariables.cpp

```diff
@@ -0,0 +1,226 @@
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesHelpers.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/DiscreteRotation.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "Domain/OrientationMap.hpp"
+#include "Domain/Side.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+using Affine = CoordinateMaps::Affine;
+using Affine2D = CoordinateMaps::ProductOf2Maps<Affine, Affine>;
+using Affine3D = CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
+
+// Simple tensor for "by hand" tests
+struct ScalarTensor {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t SpatialDim>
+struct Coords {
+  using type = tnsr::I<DataVector, SpatialDim, Frame::Inertial>;
+};
+
+void test_1d_orient_variables() noexcept {
+  const Index<1> extents{4};
+  Variables<tmpl::list<ScalarTensor>> vars(extents.product());
+  get(get<ScalarTensor>(vars)) = DataVector{{1.0, 2.0, 3.0, 4.0}};
+
+  // Check aligned case
+  {
+    const auto oriented_vars = orient_variables(vars, extents, {});
+    CHECK(oriented_vars == vars);
+  }
+
+  // Check anti-aligned case
+  {
+    const OrientationMap<1> orientation_map(
+        std::array<Direction<1>, 1>{{Direction<1>::lower_xi()}});
+    const auto oriented_vars = orient_variables(vars, extents, orientation_map);
+    Variables<tmpl::list<ScalarTensor>> expected_vars(extents.product());
+    get(get<ScalarTensor>(expected_vars)) = DataVector{{4.0, 3.0, 2.0, 1.0}};
+    CHECK(oriented_vars == expected_vars);
+  }
+}
+
+// Test one case by hand. This test case is redundant with (though not
+// identical to) one of the cases hit by `test_2d_orient_variables`. However, by
+// writing it out by hand, we provide a sanity check and a clearer example of
+// how to use `orient_variables`.
+void test_2d_simple_case_by_hand() noexcept {
+  const OrientationMap<2> orientation_map(std::array<Direction<2>, 2>{
+      {Direction<2>::lower_eta(), Direction<2>::upper_xi()}});
+  const auto extents = Index<2>{2, 3};
+
+  Variables<tmpl::list<ScalarTensor>> vars(6);
+  get(get<ScalarTensor>(vars)) = DataVector{{1.0, 2.0, 3.0, 4.0, 5.0, 6.0}};
+  const auto oriented_vars = orient_variables(vars, extents, orientation_map);
+
+  Variables<tmpl::list<ScalarTensor>> expected_vars(6);
+  get(get<ScalarTensor>(expected_vars)) =
+      DataVector{{2.0, 4.0, 6.0, 1.0, 3.0, 5.0}};
+  CHECK_ITERABLE_APPROX(get<ScalarTensor>(oriented_vars),
+                        get<ScalarTensor>(expected_vars));
```

Comment:
This should be exact, no?  (And below as well.)

Reply:
Yes, indeed it is 👍